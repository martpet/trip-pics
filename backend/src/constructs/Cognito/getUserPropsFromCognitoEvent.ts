import { PostAuthenticationTriggerEvent, PostConfirmationTriggerEvent } from 'aws-lambda';

import { CognitoIdentity, ProviderName, UserPropsFromCognitoEvent } from './cognitoTypes';

export const getUserPropsFromCognitoEvent = (
  event: PostConfirmationTriggerEvent | PostAuthenticationTriggerEvent
) => {
  const { userAttributes } = event.request;
  const identity = JSON.parse(userAttributes.identities)[0] as CognitoIdentity;

  const props: UserPropsFromCognitoEvent = {
    username: event.userName,
    sub: userAttributes.sub,
    providerName: identity.providerName as ProviderName,
    givenName: userAttributes.given_name,
    familyName: userAttributes.family_name,
    picture: userAttributes.picture,
    email: userAttributes.email,
    dateCreated: identity.dateCreated,
  };

  return props;
};
