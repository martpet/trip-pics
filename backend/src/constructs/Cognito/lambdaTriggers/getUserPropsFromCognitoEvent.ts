import { PostAuthenticationTriggerEvent, PostConfirmationTriggerEvent } from 'aws-lambda';

import { CognitoIdentity, ProviderName, UserPropsFromCognito } from '~/types';

export const getUserPropsFromCognitoEvent = (
  event: PostConfirmationTriggerEvent | PostAuthenticationTriggerEvent
): UserPropsFromCognito => {
  const { userAttributes } = event.request;
  const identity = JSON.parse(userAttributes.identities)[0] as CognitoIdentity;

  return {
    username: event.userName,
    sub: userAttributes.sub,
    providerName: identity.providerName as ProviderName,
    givenName: userAttributes.given_name,
    familyName: userAttributes.family_name,
    picture: userAttributes.picture,
    email: userAttributes.email,
    dateCreated: identity.dateCreated,
  };
};
