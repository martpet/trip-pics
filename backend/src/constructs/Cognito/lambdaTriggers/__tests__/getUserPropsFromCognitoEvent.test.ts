import { getUserPropsFromCognitoEvent } from '../getUserPropsFromCognitoEvent';
import event from './__fixtures__/postConfirmationEvent';

describe('getUserPropsFromCognitoEvent', () => {
  test('Return user props', () => {
    expect(getUserPropsFromCognitoEvent(event)).toMatchSnapshot();
  });
});
