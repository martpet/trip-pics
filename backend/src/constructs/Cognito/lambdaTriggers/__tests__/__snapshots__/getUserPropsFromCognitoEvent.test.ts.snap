// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUserPropsFromCognitoEvent Return user props 1`] = `
Object {
  "dateCreated": 1658604803064,
  "email": "name@email.com",
  "familyName": "Doe",
  "givenName": "John",
  "picture": "https://lh3.googleusercontent.com/a/asdf",
  "providerName": "Google",
  "sub": "lkjasdflk-adslfkj-asdfasd",
  "username": "someuser",
}
`;
